### ISSUE TITLE
version 1.7.0RC2 java.lang.IllegalAccessError: tried to access method org.mockito.internal.progress.ThreadSafeMockingProgress.<init>()

### ISSUE THREAD

**Issue by emmaLP** [powermock/powermock#758]
Get the following exception when trying to control constructor instantiation.

```
java.lang.IllegalAccessError: tried to access method org.mockito.internal.progress.ThreadSafeMockingProgress.<init>()V from class org.powermock.api.mockito.internal.expectation.AbstractConstructorExpectationSetup

	at org.powermock.api.mockito.internal.expectation.AbstractConstructorExpectationSetup.<init>(AbstractConstructorExpectationSetup.java:33)
	at org.powermock.api.mockito.internal.expectation.DefaultConstructorExpectationSetup.<init>(DefaultConstructorExpectationSetup.java:30)
	at org.powermock.api.mockito.PowerMockito.whenNew(PowerMockito.java:516)
```

Upon investigation this, it is caused by Mockito 2.7.12 converting constructors to private and exposing a public static method to instantiate the object.

Current class AbstractConstructorExpectationSetup in  where the error is caused:
```
    public AbstractConstructorExpectationSetup(Class<T> mockType) {
        this.mockType = mockType;
        this.mockingProgress = new ThreadSafeMockingProgress();
    }
```

This should now be changed to:
```
    public AbstractConstructorExpectationSetup(Class<T> mockType) {
        this.mockType = mockType;
        this.mockingProgress = ThreadSafeMockingProgress.mockingProgress();
    }
```

**Comment by emmaLP** [comment]
Seems this one has already been fixed in this commit https://github.com/powermock/powermock/commit/92ef1146ee05691e95dbda62d26abd18e2a76fbd#diff-e5e69931b024b775ed52fbeedc7143ee

**Comment by thekingnothing** [comment]
@emmaLP, as I understand you, I may closed the issue with resolution fixed in PowerMock 1.7.0. Am I right?
